force_wfmanager/left_side_pane/data_source_model_view.py

```diff
@@ -260,12 +260,21 @@ def update_slot_info_names(self):
                              self.output_slots_representation):
             row.name = info.name
 
-    @on_trait_change('variable_names_registry.available_variables[]')
+    @on_trait_change('variable_names_registry.available_variables[],'
+                     'output_slots_representation.name,'
+                     'output_slots_representation.type,'
+                     'input_slots_representation.name,'
+                     'input_slots_representation.type,')
     def update_data_source_input_rows(self):
-        available_variables = self._get_available_variables()
         for input_slot_row in self.input_slots_representation:
+            available_variables = self._get_available_variables(
+                input_slot_row.type)
             input_slot_row.available_variables = available_variables
 
-    def _get_available_variables(self):
+    def _get_available_variables(self, var_type=None):
```

Comment:
If the type of a input variable is empty the set of available variables does not restrict based on type. Though this should not be something which occurs the way we have things set up now, as datasource input types are not settable afaik.

Reply:
types are going to be cuba types. they will have their own verification for compatibility, so I'd take it out.